trace.go

```diff
@@ -82,7 +82,12 @@ func DebugReport(err error) string {
 	return err.Error()
 }
 
-func wrap(err error, depth int, args ...interface{}) Error {
+func wrap(err error, message string, args ...interface{}) Error {
```

Comment:
are you sure this wrap works as expected? you've just increased nesting level by 1 but still use 2